File: CinemaNow/CinemaNow.Model/DTOs/ReservationMovieDto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CinemaNow.Models.DTOs
{
    public class ReservationMovieDto
    {
        public int ReservationId { get; set; }
        public DateTime ReservationDate { get; set; }
        public int ScreeningId { get; set; }
        public DateTime ScreeningDate { get; set; }
        public List<int> SeatIds { get; set; } = new List<int>();

        public int MovieId { get; set; }
        public string? MovieTitle { get; set; }
        public int? MovieDuration { get; set; }
        public string? MovieSynopsis { get; set; }
        public string? MovieImageBase64 { get; set; }
    }

}

File: CinemaNow/UI/cinemanow_mobile/lib/models/DTOs/reservation_movie_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation_movie_dto.g.dart';

@JsonSerializable()
class ReservationMovieDto {
  final int reservationId;
  final DateTime reservationDate;
  final int screeningId;
  final DateTime screeningDate;
  final List<int> seatIds;
  final int movieId;
  final String? movieTitle;
  final int? movieDuration;
  final String? movieSynopsis;
  final String? movieImageBase64;

  ReservationMovieDto({
    required this.reservationId,
    required this.reservationDate,
    required this.screeningId,
    required this.screeningDate,
    required this.seatIds,
    required this.movieId,
    this.movieTitle,
    this.movieDuration,
    this.movieSynopsis,
    this.movieImageBase64,
  });

  factory ReservationMovieDto.fromJson(Map<String, dynamic> json) =>
      _$ReservationMovieDtoFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationMovieDtoToJson(this);
}
